sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -588,7 +639,7 @@ case class AdaptiveSparkPlanExec(
       if (plan.children.isEmpty) {
         CreateStageResult(newPlan = plan, allChildStagesMaterialized = true, newStages = Seq.empty)
       } else {
-        val results = plan.children.map(createQueryStages)
+        val results = plan.children.map(createQueryStagesInternal)
         CreateStageResult(
           newPlan = plan.withNewChildren(results.map(_.newPlan)),
           allChildStagesMaterialized = results.forall(_.allChildStagesMaterialized),
```

Comment:
It seems the new code is a bit hard to read. Not sure if there are some developing context.

Can we create result query stage here ? If the plan is root query and `allChildStagesMaterialized` then we wrap `ResultQueryStage` and it is not a materialized stage, so aqe will materialize it.

Reply:
sounds reasonable to me, cc @liuzqt